src/client.js

```diff
@@ -0,0 +1,30 @@
+import Dashcore from '@dashevo/dashcore-lib'
+
+export default class Client {
+  constructor (socket) {
+    this.socket = socket
+  }
+
+  #modifiedTransaction (tx) {
+    return {
+      txId: tx.hash,
+      version: tx.version,
+      type: tx.type,
+      size: tx.toBuffer().length,
+      locktime: tx.nLockTime,
+      from: tx.inputs.map(input => input.script?.toAddress(Dashcore.Networks.testnet).toString()),
+      to: tx.outputs.map(output => {
+        return {
+          address: output.script?.toAddress(Dashcore.Networks.testnet).toString(),
+          valueSat: output.satoshis
+        }
+      })
+    }
+  }
+
+  broadcastMessage (message) {
+    if (this.socket.readyState === 1) {
+      this.socket.send(JSON.stringify(this.#modifiedTransaction(message)))
```

Comment:
You should not map here I believe, because the function is intended to just send message to the client, it should just take the input and send it over the socket. The mapping should occur on the upper level